include/deal.II/grid/grid_generator.h

```diff
@@ -28,6 +28,8 @@
 #include <deal.II/grid/reference_cell.h>
 #include <deal.II/grid/tria.h>
 
+#include <deal.II/cgal/triangulation.h>
+
```

Comment:
I'm really sorry @drwells but there's something I don't get. You suggest to add header and source files `additional_data.h,additional_data.cc` declaring and defining `AdditionalData,` respectively. 
Then, I don't understand what you would put in this line in place of `#include <deal.II/cgal/triangulation.h>`. 

With `#include <deal.II/cgal/additional_data.h>` this won't compile, unless I put all CGAL headers needed by `implicit_function()` inside `additional_data.h`.

Reply:
Not a problem - the point of this PR is to move the implementation into `grid_generator.cc`. We can include all the necessary CGAL headers there instead.

Reply:
Thanks. I've done as suggested, even if I defined everything inside `additional_data.h`, since the CGALWrappers are defined directly inside headers. 
When compiling, I only get a `warning: unused variable ‘extent’` in instantiation of ‘void dealii::GridGenerator::replicate_triangulation()`

Reply:
I don't know about that message - that doesn't seem related.